scripts/tests/test-ensure-frozen-migrations.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+set -euo pipefail
+set -x
+
+script_path='scripts/ensure-frozen-migrations'
+schema_dir='backend/controller/sql/schema'
+
+function create() {
+  echo "CREATE TABLE testing (id SERIAL PRIMARY KEY);" > "$schema_dir/testing.sql"
+}
+
+function remove_existing() {
+  rm "$schema_dir/20240704103403_create_module_secrets.sql"
+}
+
+function modify_comment_in_existing() {
+  sed -i.bak 's/Function for deployment notifications./🧐/' "$schema_dir/20231103205514_init.sql"
+  rm "$schema_dir/20231103205514_init.sql.bak"
+}
+
+function modify_content_in_existing() {
+  sed -i.bak 's/CREATE OR REPLACE FUNCTION notify_event()/CREATE 🥲 notify_event()/' "$schema_dir/20231103205514_init.sql"
+  rm "$schema_dir/20231103205514_init.sql.bak"
+}
+
+function no_changes() {
+  echo "no-changes" >> .gitignore
+  git add .gitignore
+}
+
+function commit_migrations_dir() {
+  git add "$schema_dir"
+  git commit -m "ci: automated test commit, this should not be pushed!"
+}
+
+
+# higher order test function, accepting a function as an argument, and an expected fail or pass.
+# omitted means it should pass.
+# fail should be used when the test is expected to fail.
+function test() {
```

Comment:
Isn't this going to mutate the working directory? That's not going to be great to run through lefthook. Probably better to clone it into a tmp dir first.

Reply:
Yes it will--I did put in effort to make sure the test cleans up, without messing with any staged code, but looking over it again, if the dev has code in migrations that isn't committed it will probably get nuked.

I think rather than another clone, I'll get rid of if from lefthook since it is in CI and the script doesn't seem like it will need that much change in the future to warrant every push to be tested locally (plus the risk of code nukage!)